src/providers/retryProvider.ts

```diff
@@ -260,11 +260,29 @@ export class RetryProvider extends ethers.providers.StaticJsonRpcProvider {
     return response;
   }
 
-  _trySend(provider: ethers.providers.StaticJsonRpcProvider, method: string, params: Array<unknown>): Promise<unknown> {
+  async _trySend(
+    provider: ethers.providers.StaticJsonRpcProvider,
+    method: string,
+    params: Array<unknown>
+  ): Promise<unknown> {
+    const loop = true;
     let promise = this._sendAndValidate(provider, method, params);
-    for (let i = 0; i < this.retries; i++) {
-      promise = promise.catch(() => delay(this.delay).then(() => this._sendAndValidate(provider, method, params)));
-    }
+    let i = 0;
+
+    do {
+      try {
+        await promise;
+        break;
+      } catch (err: unknown) {
+        if (++i >= this.retries) {
+          throw err;
```

Comment:
I like the suggestion - it's cleaner and more concise than what I had. At this point I think we only care about the final error, but we might iterate on this a bit and start exposing the underlying errors. It might be noisy, though.

https://github.com/across-protocol/sdk/pull/761/commits/d0a877df2808af58d5218f277dcfcc08b2da88d4